### DISCUSSION TITLE
Thoughts on Release of Version 0.4

### DISCUSSION THREAD

**Discussion by Avnsx** [Avnsx/fansly#100] [Announcements]
Introducing the compiled Windows-exclusive version of Fansly Downloader, offering an executable file that allows you to conveniently launch the program without the need for Python installation. To begin, please unzip the zip file using WinRAR or a similar program.

⚠️ The "updater.exe" file is included solely for backward compatibility with version 0.3.5. It enables the previous version to update to version 0.4. However, if you're manually downloading version 0.4 from the repository, you can safely delete the "updater.exe" file, as the self-updating feature is now integrated within Fansly Downloader itself. ⚠️

Read [Quick Start](https://github.com/Avnsx/fansly-downloader#-quick-start) to know how to run Fansly Downloader

# Changelog v0.4
```
changes:
- re-wrote basically the whole codebase
- re-written versions of configurator & updater are now integrated into the downloader
- option naming_convention is now deprecated & gets removed automatically
- config.ini input values are now actually being validated for correctness
    - e.g. all important config values will now be interactively set-up with user
- update_recent_download has been improved drastically
    - now only supports Auto or False
    - its enabled by default now
    - it will be way quicker, by utilising deduplication
    - deduplication is done through media id & hashing now
    - hashes are added to filenames initially
    - filenames with hashes, will not get hashed again
    - update_recent_download removed from config file, will silently run by default
- content upload timestamps are now converted into local systems timezone
- adapted fansly scraper for the new m3u8 video format
    - this fixes the 720p video downloads from previous version
- fansly scraper is now way more performant and efficient in every aspect
- you can now also download media collections and posts by post ID
- greatly improved cross-platform compatibility

fixed issues:
- fixed a issue with not escaping apostrophes on macOS (issues id 78)
- fixed subscriber content missing (issues id 59)
- all media file extensions are now supported (issues id 75)
- where content would falsely get flagged as preview (issues id 91)
- where a TypeError would be caused (issues id 90)
- where m3u8 videos would transcode with a 2 second delay at the start (issues id 89)
- fansly downloader will now actually get the true highest quality of media
- fixed a bug where videos would get recognised as images
- audio will be downloaded as mp3 now
```

Full Changelog: https://github.com/Avnsx/fansly/compare/v0.3.5...v0.4

Read about [configuration options](https://github.com/Avnsx/fansly-downloader/wiki/Explanation-of-provided-programs-&-their-functionality#explanation-of-configini)

VirusTotal ([detections?](https://github.com/Avnsx/fansly-downloader#-faq)):
Fansly Downloader.exe: https://rebrand.ly/downloader_04

Compiled using pyinstaller on python 3.8.10

<hr /><em>This discussion was created from the release <a href='https://github.com/Avnsx/fansly-downloader/releases/tag/v0.4'>v0.4: bug fix</a>.</em>

**Comment by crunk323** [comment]
Thank you for all the work you put into this. Everything is working fine for me so far. I'll update if I come across any issues while I catch up on all the content I missed out on saving this past month.
